src/index.js

```diff
@@ -1,73 +1,72 @@
+// main.js
 import './style.css';
+import { saveTasksToLocalStorage, updateIndexes } from './modules/taskFunctions.js';
+import { addTaskToList, updateTaskTextDecoration, deleteSelectedTasks } from './modules/taskListFunctions.js';
+import { enableTaskEditing, saveTaskEdit } from './modules/editFunctions.js';
 
 const taskForm = document.getElementById('task-form');
 const taskInput = document.getElementById('task-input');
 const taskList = document.getElementById('task-list');
 const deleteSelectedButton = document.getElementById('delete-selected');
 
-const hardcodedTasks = [
-  { text: 'Buy Groceries', completed: false },
-  { text: 'Complete ToDo List Project', completed: true },
-  { text: 'Do coding challenges', completed: false },
-];
-
-function saveTasksToLocalStorage() {
-  const tasks = [];
+// Save data to local storage on page load
+const savedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
+savedTasks.forEach((savedTask) => {
+  addTaskToList(taskList, savedTask.text);
   const taskItems = taskList.querySelectorAll('li');
-  taskItems.forEach((taskItem) => {
-    const checkbox = taskItem.querySelector("input[type='checkbox']");
-    const text = taskItem.querySelector('span').textContent;
-    const completed = checkbox.checked;
-    tasks.push({ text, completed });
-  });
-  localStorage.setItem('tasks', JSON.stringify(tasks));
-}
-
-function displayTasks() {
-  hardcodedTasks.forEach((task) => {
-    const li = document.createElement('li');
-    li.innerHTML = `
-      <input type="checkbox" ${task.completed ? 'checked' : ''}>
-      <span>${task.text}</span>
-    `;
-    taskList.appendChild(li);
-  });
-}
-
-displayTasks();
+  const checkbox = taskItems[taskItems.length - 1].querySelector("input[type='checkbox']");
+  if (savedTask.completed) {
+    checkbox.checked = true;
+    updateTaskTextDecoration(checkbox, savedTask.text);
+  }
+});
 
 taskForm.addEventListener('submit', (event) => {
   event.preventDefault();
 
   const taskText = taskInput.value.trim();
   if (taskText === '') return;
 
-  const li = document.createElement('li');
-  li.innerHTML = `
-    <input type="checkbox">
-    <span>${taskText}</span>
-  `;
-  taskList.appendChild(li);
+  addTaskToList(taskList, taskText);
 
   taskInput.value = '';
 
-  saveTasksToLocalStorage();
+  saveTasksToLocalStorage(taskList);
 });
 
 taskList.addEventListener('change', (event) => {
   if (event.target.tagName === 'INPUT' && event.target.type === 'checkbox') {
     const taskText = event.target.nextElementSibling;
-    taskText.style.textDecoration = event.target.checked ? 'line-through' : 'none';
-    saveTasksToLocalStorage();
+    updateTaskTextDecoration(event.target, taskText);
+    saveTasksToLocalStorage(taskList);
   }
 });
 
 deleteSelectedButton.addEventListener('click', () => {
```

Comment:
![image](https://github.com/DevKaranJ/ToDo-list/assets/93039359/9f6e4cd1-7282-4307-9e55-6c328e860caf)

- [ ] It would be nice if you made delete buttons for each task as you can see in the given design of microverse 👍 but I saw that you have implemented the clear task button but this milestone expect a dedicated delete button for each task or todo 👍